shared/actions/teams.js

```diff
@@ -394,38 +394,51 @@ const _getDetails = function*(action: TeamsGen.GetDetailsPayload): Saga.SagaGene
     }, {})
 
     const infos = []
-    const types = ['admins', 'owners', 'readers', 'writers']
-    const typeMap = {
-      admins: 'admin',
-      owners: 'owner',
-      readers: 'reader',
-      writers: 'writer',
+    const types: Types.TeamRoleType[] = ['reader', 'writer', 'admin', 'owner']
```

Comment:
Doing it this way types better, since the type we pass to `makeMemberInfo` is guaranteed non-null